core/server/src/main/java/alluxio/proxy/PathsRestServiceHandler.java

```diff
@@ -0,0 +1,375 @@
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.proxy;
+
+import alluxio.AlluxioURI;
+import alluxio.RestUtils;
+import alluxio.client.file.FileInStream;
+import alluxio.client.file.FileOutStream;
+import alluxio.client.file.FileSystem;
+import alluxio.client.file.URIStatus;
+import alluxio.client.file.options.CreateDirectoryOptions;
+import alluxio.client.file.options.CreateFileOptions;
+import alluxio.client.file.options.DeleteOptions;
+import alluxio.client.file.options.ExistsOptions;
+import alluxio.client.file.options.FreeOptions;
+import alluxio.client.file.options.GetStatusOptions;
+import alluxio.client.file.options.ListStatusOptions;
+import alluxio.client.file.options.MountOptions;
+import alluxio.client.file.options.OpenFileOptions;
+import alluxio.client.file.options.RenameOptions;
+import alluxio.client.file.options.SetAttributeOptions;
+import alluxio.client.file.options.UnmountOptions;
+import alluxio.web.ProxyWebServer;
+
+import com.google.common.base.Preconditions;
+import com.qmino.miredot.annotations.ReturnType;
+
+import java.util.List;
+
+import javax.annotation.concurrent.NotThreadSafe;
+import javax.servlet.ServletContext;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+
+/**
+ * This class is a REST handler for file system client requests.
+ */
+@NotThreadSafe
+@Path(PathsRestServiceHandler.SERVICE_PREFIX)
+@Produces(MediaType.APPLICATION_JSON)
+@Consumes(MediaType.APPLICATION_JSON)
+public final class PathsRestServiceHandler {
+  public static final String SERVICE_PREFIX = "paths";
+
+  public static final String PATH_PARAM = "{path:.*}/";
+
+  public static final String CREATE_DIRECTORY = "create-directory";
+  public static final String CREATE_FILE = "create-file";
+  public static final String DELETE = "delete";
+  public static final String EXISTS = "exists";
+  public static final String FREE = "free";
+  public static final String GET_STATUS = "get-status";
+  public static final String LIST_STATUS = "list-status";
+  public static final String MOUNT = "mount";
+  public static final String OPEN_FILE = "open-file";
+  public static final String RENAME = "rename";
+  public static final String SET_ATTRIBUTE = "set-attribute";
+  public static final String UNMOUNT = "unmount";
+
+  private final FileSystem mFileSystem;
+
+  /**
+   * Constructs a new {@link PathsRestServiceHandler}.
+   *
+   * @param context context for the servlet
+   */
+  public PathsRestServiceHandler(@Context ServletContext context) {
+    mFileSystem =
+        (FileSystem) context.getAttribute(ProxyWebServer.FILE_SYSTEM_SERVLET_RESOURCE_KEY);
+  }
+
+  /**
+   * @summary creates a directory
```

Comment:
is there a place where we can write a full description that will go into the miredot docs?